modules/heat_conduction/include/meshgenerators/PatchSidesetGenerator.h

```diff
@@ -0,0 +1,50 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "MeshGenerator.h"
+#include "libmesh/replicated_mesh.h"
+
+// Forward declarations
+class PatchSidesetGenerator;
+
+template <>
+InputParameters validParams<PatchSidesetGenerator>();
+
+/**
+ * Subdivides a sidesets into smaller patches each of which is going
+ * to be a new patch
+ */
+class PatchSidesetGenerator : public MeshGenerator
+{
+public:
+  PatchSidesetGenerator(const InputParameters & parameters);
+
+  std::unique_ptr<MeshBase> generate() override;
+
+protected:
+  /// returns the name of the _n_patches subdivisions derived from _sideset
+  std::vector<BoundaryName> sidesetNameHelper(const std::string & base_name) const;
+
+  Elem * boundaryElementHelper(std::unique_ptr<libMesh::ReplicatedMesh> & mesh, int type) const;
```

Comment:
Style comment: Here you aren't intending to pass ownership of the mesh, which is why you are passing a reference to a smart pointer. Since you know that the mesh always exists (e.g. you aren't checking for `nullptr`) there's really no reason to muddy up the waters with this complicated construct. Just pass the mesh by reference (e.g. have the callee dereference the smart pointer and change these declarations to be just `libMesh::ReplicatedMesh & mesh` types).